### PR TITLE
Use satellite icechunk instead of zipped zarr

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

This PR updates the satellite IO to pull the satellite data from the icechunk store instead of the `latest.zarr.zip`

## How Has This Been Tested?

I have successfully run the app locally to make predictions

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: scripts/run_app_local_example.ipynb
Comment (dfulu) [inline scripts/run_app_local_example.ipynb:1]: This was very out of date, so I deleted it instead of updating the env var and leaving it otherwise broken

File: README.md [hunk 1]
```diff
@@ -59,21 +54,6 @@ Forecasts and GSP capacities are read from and written to the data platform.
   will raise an exception if any model fails. If set to "critical" it will raise an exception if any
   critical model fails. If not set, it will not raise an exception.
 
-### Examples
```
Comment (dfulu) [inline README.md:62]: I deleted this rather than updating since it seemed unnecessary

File: tests/integration/test_app.py [hunk 2]
```diff
@@ -98,55 +100,21 @@ async def test_app(
...
+            side_effect=[sat_5_data_zero_delay],
+        ):  
+            # Run prediction
+            await run(t0=test_t0)
 
     model_configs = get_all_models(get_critical_only=False)
-    await check_number_of_forecasts(client, model_configs, test_t0)
-
-
-@pytest.mark.asyncio(loop_scope="session")
```
Comment (dfulu) [inline tests/integration/test_app.py:119]: We now assume there will always be an icechunk. But it can be out of date

File: tests/data/test_satellite.py [hunk 3]
```diff
@@ -51,114 +32,24 @@ def timesteps_match_expected_freq(sat_path: str, expected_freq_mins: int | list[
...
-
 def test_run_sat_5_data(sat_5_data, test_t0):
     """Download and process only the 5 minute satellite data"""
 
     with tempfile.TemporaryDirectory() as tmpdirname:
         os.chdir(tmpdirname)
 
-        # Make 5-minutely satellite data available
-        save_to_zarr_zip(sat_5_data, filename="latest.zarr.zip")
+        with patch("pvnet_app.data.satellite.open_satellite_data", side_effect=[sat_5_data]):
```
Comment (dfulu) [inline tests/data/test_satellite.py:41]: We mock this internal function rather than making a code path to load from a local icechunk just for the sake of the tests

File: src/pvnet_app/data/satellite.py [hunk 4]
```diff
@@ -483,14 +413,39 @@ def resave(self, ds: xr.Dataset) -> None:
...
+
+        if self.source_path_15 is not None:
+            # Also open 15-minute satellite
+            ds_dict["15-min"] = open_satellite_data(
+                s3_icechunk_path=self.source_path_15,
+                region=self.s3_region,
+            )
+
+        # Select the source with the most recent data
+        best_source = max(ds_dict, key=lambda k: ds_dict[k].time.max())
```
Comment (peterdudfield) [inline src/pvnet_app/data/satellite.py]: how do you know if it loads 5 or 15, if the max time is the same?

Comment (dfulu) [reply src/pvnet_app/data/satellite.py]: Good point. I'll make that more explicit

File: src/pvnet_app/app.py [hunk 5]
```diff
@@ -187,6 +190,7 @@ async def run(
             t0=t0,
             source_path_5=sat_source_path_5,
             source_path_15=sat_source_path_15,
+            s3_region=sat_s3_region,
```
Comment (peterdudfield) [inline src/pvnet_app/app.py:193]: do you need to get the time_window from the configurations at atll? Incase we use more than 1 hour of satellite data?

File: src/pvnet_app/save.py [hunk 6]
```diff
@@ -341,10 +341,10 @@ async def get_metadata_for_forecast(
...
     # add nwp and satellite last updated time, load file from s3 if exists
-    env_vars = ["NWP_ECMWF_ZARR_PATH",
-                "NWP_UKV_ZARR_PATH",
-                "SATELLITE_ZARR_PATH",
-                "SATELLITE_15_ZARR_PATH"]
+    env_vars = [
+        "NWP_ECMWF_ZARR_PATH",
+        "NWP_UKV_ZARR_PATH",
+    ]
     for env_var in env_vars:
```
Comment (peterdudfield) [inline src/pvnet_app/save.py]: we need to still add the `satellite_last_modified` (and 15) into the metadata, as this is what clients see when the data is last updated. You might have to get this from s3

File: tests/integration/test_app.py (hunk 2 above)
Comment (peterdudfield) [reply tests/integration/test_app.py:119]: are we sure we want to get rid of this? I think this has caught us out in the past. 

I understand we want to keep the code clean vs make sure it works for all edge cases. I would be tempted to make sure this behaviour still works

Comment (dfulu) [reply tests/integration/test_app.py:119]: What would the test be? Make sure it still runs if the icechunk doesn't exist?

File: src/pvnet_app/save.py (hunk 6 above)
Comment (dfulu) [reply src/pvnet_app/save.py]: Okay, I'll need to find a way to grab that from the icechunk. 

Though looking at this, I was wondering if it would make more sense for us to store the NWP init-times and the time of the last satellite timestamp? They seem like the more relevant pieces of information rather than the last time we edited the file? For the satellite images it is possible/likely that the images available from EUMETSAT lag the time we updated them

File: tests/integration/test_app.py (hunk 2 above)
Comment (peterdudfield) [reply tests/integration/test_app.py:119]: something like that,

File: src/pvnet_app/app.py (hunk 5 above)
Comment (dfulu) [reply src/pvnet_app/app.py:193]: I left the `time_window` only partially plumbed through since we don't currently have a use-case to change it. I think its very unlikely we ever need more than 1 hour but its possible we could drop it to 30 mins in the future.

I think it's best to leave it for now, we can easily edit it in the future if we need to. And to make it depend on config settings across all the models would require us to do a little bit of a refactor of how we do this now, so I don't think it's worth the work and code churn

File: src/pvnet_app/save.py (hunk 6 above)
Comment (peterdudfield) [reply src/pvnet_app/save.py]: Both are useful, currerntly the users want to know when was the last time the nwp and the satellite were updated

Comment (dfulu) [reply src/pvnet_app/save.py]: But if the update time was T are the users assuming we have satellite images up to T. That isn't necessarily the case. If instead we stored the latest timestamp that would be more explicit

Comment (dfulu) [reply src/pvnet_app/save.py]: I've reinstated it as it was

**Review by peterdudfield — APPROVED** [review]
